classes/class-backup.php

```diff
@@ -1828,6 +1839,37 @@ public function error_handler( $type ) {
 
 		}
 
+		/**
+		 * Determine if user can connect via the CLI
+		 *
+		 * @return \WP_Error
+		 */
+		public function user_can_connect() {
+
+			// mysql --host=localhost --user=myname --password=mypass mydb
+
+			$user = escapeshellarg( DB_USER );
+
+			$pwd = escapeshellarg( DB_PASSWORD );
+
+			$host = escapeshellarg( DB_HOST );
+
+			$db = escapeshellarg( DB_NAME );
+
+			$cmd = "mysql --host={$host} --user={$user} --password={$pwd} {$db}";
+
+			// Pipe STDERR to STDOUT
+			$cmd .= ' 2>&1';
+
+			// Store any returned data in an error
+			$stderr = shell_exec( $cmd );
+
+			if ( $stderr ) {
```

Comment:
I think we will always get an error on MySQL > 5.6 because we are passing password via CLI (like here: https://github.com/humanmade/backupwordpress/pull/698/files#diff-0c135420eb3271449b66b83a8aa89035R880